privacyidea/lib/token.py

```diff
@@ -2690,3 +2692,33 @@ def token_load(token_dict, tokenowner=True, overwrite=False):
                 raise TokenAdminError("Token can not be assiend to the specified user!")
 
     return token
+
+
+def challenge_text_replace(message, user, token_obj):
+    serial = token_obj.token.serial if token_obj.token.serial else None
+    tokenowner = user if user else None
+    tokentype = token_obj.token.tokentype if token_obj.token.tokentype else ""
+    tags = create_tag_dict(serial=serial, tokenowner=tokenowner, tokentype=tokentype)
+
+    if tokentype == "sms":
+        if is_true(token_obj.get_tokeninfo("dynamic_phone")):
+            phone = token_obj.user.get_user_phone("mobile")
+            if type(phone) == list and phone:
+                # if there is a non-empty list, we use the first entry
+                phone = phone[0]
+        else:
+            phone = token_obj.get_tokeninfo("phone")
```

Comment:
is this guaranteed to return not none? Otherwise a check before adding it to the tags would be necessary

Reply:
https://github.com/privacyidea/privacyidea/pull/4028/commits/08943710ba4c55d902d2f0e7fc54632cc2724e3f
I think it is not necessary but to be sure I added the check.